### PR TITLE
Fix TestQuerierStreamingRemoteRead flakyness

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

@dimitarvdimitrov reported me that `TestQuerierStreamingRemoteRead` is flaky after recent changes. I reproduce it locally. The problem is that TSDB chunks cutting is based on a estimation, the estimation is computed based on samples timestamp, and due to rounding the chunks may be cut at slightly different times. The easiest way to fix flakyness is by simply running the test always with the same mocked time, which is what I'm doing in this PR.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: integration/querier_remote_read_test.go [hunk 1]
```diff
@@ -333,7 +337,6 @@ func TestQuerierStreamingRemoteRead(t *testing.T) {
 					require.NoError(t, err)
 
 					chkItr := chk.Iterator(nil)
-					chkIdx := 0
```
Comment (pracucci) [inline integration/querier_remote_read_test.go:336]: Note to reviewers: removed because unused.

**Review by krajorama — APPROVED** [review]
LGTM

File: integration/querier_remote_read_test.go [hunk 2]
```diff
@@ -152,7 +152,9 @@ func runTestPushSeriesForQuerierRemoteRead(t *testing.T, c *e2emimir.Client, que
 
 func TestQuerierStreamingRemoteRead(t *testing.T) {
 	var (
-		now           = time.Now().Truncate(time.Minute) // Make assertions easier to debug.
+		// This test runs with a fixed time so that assertions are stable. The reason is that the formula used by TSDB
+		// to cut chunks has some rounding which may cause this test to be flaky if run with a random "now" time.
+		now           = must(time.Parse(time.RFC3339, "2024-01-01T00:00:00Z"))
```
Comment (pracucci) [inline integration/querier_remote_read_test.go:157]: Note to reviewers: I intentionally picked an "old" timestamp to ensure the test can successfully run.

**Review by dimitarvdimitrov — COMMENTED** [review]
thanks

**Pull request merged**